include/sempr/entity/spatial/PointCloud.hpp

```diff
@@ -216,12 +218,18 @@ class PointCloud : public Geometry, public AbstractPointCloud<double> //ToDo Cha
     {
         // Note: this is an ineffective way to it because for each call it will create a copy of the coordinate on the heap!
         // But this version is safe! Dynamicly type remapping could easily cause undefined behaviour.
-        return std::make_shared<CoordinatePoint>(*getGeometry()->getGeometryN(idx)->getCoordinate());
+        // So far a const cast is used to get writeable access to the coordinate without any outstanding programming effort (like filters)
+        return std::make_shared<CoordinatePoint>(*const_cast<geom::Coordinate*>(getGeometry()->getGeometryN(idx)->getCoordinate()));
     }
 
     const CoordinatePoint at(std::size_t idx) const
     {
-        return CoordinatePoint(*getGeometry()->getGeometryN(idx)->getCoordinate());
+        return CoordinatePoint(*const_cast<geom::Coordinate*>(getGeometry()->getGeometryN(idx)->getCoordinate()));
```

Comment:
Why the const-cast? Did the previous version (without it) not compile?

Reply:
This depends on the changes of the CoordinatePoint.
In the previous version the CoordinatePoint makes the const-cast and I removed it to avoid a hidden or unexpected const-cast.

Reply:
Ah, okay! Well then, feel free to merge. :)